.github/workflows/test.yaml

```diff
@@ -0,0 +1,607 @@
+        shell: bash
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf ${{ matrix.metadata.llvm_url }} -L -o llvm.tar.xz
+          LLVM_DIR=$(pwd)/${{ env.LLVM_DIR }}
+          mkdir ${LLVM_DIR}
+          tar xf llvm.tar.xz --strip-components=1 -C ${LLVM_DIR}
+          echo "${LLVM_DIR}/bin" >> $GITHUB_PATH
+          echo "LLVM_SYS_120_PREFIX=${LLVM_DIR}" >> $GITHUB_ENV
+        env:
+          LLVM_DIR: .llvm
+      - name: Setup Rust target
+        shell: bash
+        run: |
+          mkdir -p .cargo
+          cat << EOF > .cargo/config.toml
+          [build]
+          target = "${{ matrix.metadata.target }}"
+          EOF
+        if: matrix.metadata.target
+      - name: Cache
+        uses: whywaita/actions-cache-s3@v2
+        with:
+          path: |
+            ~/.cargo/*
+            ./target/*
+          key: s3-${{ github.repository }}-${{ runner.os }}-${{ hashFiles('Cargo.lock') }}-wasmer-make-build-wasmer-${{ matrix.build-what.key }}-${{ matrix.metadata.build }}
+          aws-s3-bucket: wasmer-rust-artifacts-cache
+          aws-access-key-id: ${{ secrets.CLOUDFLARE_ARTIFACTS_CACHE_ACCESS_TOKEN }}
+          aws-secret-access-key: ${{ secrets.CLOUDFLARE_ARTIFACTS_CACHE_ACCESS_KEY }}
+          aws-region: auto
+          aws-endpoint: https://1541b1e8a3fc6ad155ce67ef38899700.r2.cloudflarestorage.com
+          aws-s3-bucket-endpoint: false
+          aws-s3-force-path-style: true
+      - name: Build ${{ matrix.build-what.key }}
+        shell: bash
+        # tests were disabled for linux-musl
+        if: ${{ !(matrix.build-what.key == 'capi' && matrix.metadata.build == 'linux-musl') }}
+        run: |
+          ${{ matrix.build-what.build-cmd }}
+      - name: Build ${{ matrix.build-what.key }}
+        shell: bash
+        if: matrix.build-what.key == 'capi' && matrix.metadata.build == 'linux-musl'
+        run: |
+          make build-capi && make package-capi && make tar-capi
+      - name: Archive production artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: wasmer-cli-${{ matrix.metadata.build }}
+          path: build-wasmer.tar.gz
+          if-no-files-found: ignore
+          retention-days: 2
+      - name: Archive production artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: capi-${{ matrix.metadata.build }}
+          path: build-capi.tar.gz
+          if-no-files-found: ignore
+          retention-days: 2
+
+  test: 
+    name: ${{ matrix.stage.description }} on ${{ matrix.metadata.build }}
+    runs-on: ${{ matrix.metadata.os }}
+    needs: setup
+    strategy:
+      fail-fast: false
+      matrix:
+        stage: [
+          {
+            stage: 0,
+            description: 'Test compilers', 
+          },
+          {
+            stage: 1, 
+            description: 'cargo test --all',
+          },
+          {
+            stage: 2, 
+            description: 'Test compiler-cranelift',
```

Comment:
> The only reason I see is to make sure the compilers work fine without the other ones being compiled

That's the reason, yeah. Not sure how to move forward on that, should we continue to do this or not?

Reply:
Probably best to ask @ptitSeb if it makes sense or not.

Reply:
Not sure it's worth the work. I would not pursuit this unless there are some other advantage

Reply:
Okay, so I'm going to merge this into one test with `--compilers cranelift,singlepass + llvm?`

Reply:
Oh I just saw why we do this: because the dependencies are different when we test `--manifest-path lib/compiler-cranelift` vs `--manifest-path lib/compiler-singlepass`. I'm going to leave this as is because of this.